nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestColdStorage.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.ecm.core;
+
+import static javax.servlet.http.HttpServletResponse.SC_CONFLICT;
+import static javax.servlet.http.HttpServletResponse.SC_NOT_FOUND;
+import static org.junit.Assert.*;
+
+import java.io.IOException;
+import java.io.Serializable;
+
+import javax.inject.Inject;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.*;
+import org.nuxeo.ecm.core.blob.ColdStorageHelper;
+import org.nuxeo.ecm.core.schema.FacetNames;
+import org.nuxeo.ecm.core.test.ColdStorageFeature;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features(ColdStorageFeature.class)
+public class TestColdStorage {
+
+    protected static final String FILE_CONTENT = "foo";
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected TransactionalFeature transactionalFeature;
+
+    @Test
+    public void shouldMoveBlobDocumentToColdStorage() throws IOException {
+        DocumentModel documentModel = createDocument(true);
+
+        // move the blob to cold storage
+        documentModel = ColdStorageHelper.moveContentToColdStorage(session, documentModel.getRef());
+        transactionalFeature.nextTransaction();
+        documentModel.refresh();
+
+        assertTrue(documentModel.hasFacet(FacetNames.COLD_STORAGE));
+
+        assertNull(documentModel.getPropertyValue(ColdStorageHelper.FILE_CONTENT_PROPERTY));
+
+        // check if the `coldstorage:coldContent` contains the original file content
+        checkBlobContent(documentModel, ColdStorageHelper.COLD_STORAGE_CONTENT_PROPERTY, FILE_CONTENT);
```

Comment:
Since we have a test contribution to dispatch a blob given it's xpath, could we add assertions to verify their blob provider?
Something based ont the blob keys, similar to what is done in `TestBlobDispatcher`.
file:content => default, coldstorage:coldContent => dummy

Reply:
initially I tested it, but after that i removed it, because I told myself the test part is the responsibility of the dispatcher and if we do the right configuration it will work.
but i can add the check on the `dummy` provider:
` assertEquals("dummy", ((ManagedBlob) coldContent).getProviderId());`

for the main content as it will be `null` i think we can avoid adding the assertion.

Reply:
The main content will have a blob provider id `default` in theory (or `test` I'm not sure if it takes the repo name)